peps/pep-0747.rst

```diff
@@ -711,12 +735,38 @@ assigned to variables and manipulated like any other data in a program:
 ``TypeExpr[]`` is how you spell the type of a variable containing a
 type annotation object describing a type.
 
-``TypeExpr[]`` is similar to ``type[]``, but ``type[]`` can only used to
+``TypeExpr[]`` is similar to ``type[]``, but ``type[]`` can only
 spell simple **class objects** like ``int``, ``str``, ``list``, or ``MyClass``.
 ``TypeExpr[]`` by contrast can additionally spell more complex types, 
 including those with brackets (like ``list[int]``) or pipes (like ``int | None``),
 and including special types like ``Any``, ``LiteralString``, or ``Never``.
 
+A ``TypeExpr`` variable looks similar to a ``TypeAlias`` definition, but
```

Comment:
What is a "`TypeExpr` variable"? I think what you mean is that a variable (or parameter) can be statically evaluated to have a type of `TypeExpr[T]`?

By "`TypeAlias` definition", are you talking about a statement of the form `<name>: TypeAlias = <type expression>`, as define din PEP 613? I'm not sure how this is related to `TypeExpr`.

Perhaps you're talking about PEP 484 type aliases that have the syntactic form `<name> = <expression>` and numerous (undocumented) semantic rules and heuristics that distinguish it from a regular variable assignment? If that's the case, then I agree there's potential overlap with the `TypeExpr` concept. In particular, I was thinking that we could leverage the definitions in this PEP to (at last!) formalize the rules for PEP 484 type aliases. I'm now less sure of this given some of the other limitations we've needed to add to this PEP, such as the requirement that certain ambiguous forms must use an explicit `TypeExpr` constructor call.

Reply:
I'll alter the original line with inline examples like:

> A ``TypeExpr`` variable (`maybe_float: TypeExpr`) looks similar to a ``TypeAlias`` definition (`MaybeFloat: TypeAlias`) ...

Hopefully also the examples following the paragraph help clarify the meaning.